src/main/java/org/reaktivity/nukleus/maven/plugin/internal/generate/StringFlyweightGenerator.java

```diff
@@ -47,322 +46,148 @@ public StringFlyweightGenerator(
-                .endControlFlow()
-                .addStatement("int length0 = length0()")
-                .beginControlFlow("if (length0 != -1)")
-                .addStatement("valueRO.wrap(buffer, offset + FIELD_SIZE_LENGTH, length0)")
-                .endControlFlow()
-                .addStatement("return this")
-                .build();
-    }
-
-    private MethodSpec wrapMethod()
-    {
-        return methodBuilder("wrap")
-                .addAnnotation(Override.class)
-                .addModifiers(PUBLIC)
-                .addParameter(DIRECT_BUFFER_TYPE, "buffer")
-                .addParameter(int.class, "offset")
-                .addParameter(int.class, "maxLimit")
-                .returns(thisName)
-                .addStatement("super.wrap(buffer, offset, maxLimit)")
-                .addStatement("checkLimit(offset + FIELD_SIZE_LENGTH, maxLimit)")
-                .addStatement("checkLimit(limit(), maxLimit)")
-                .addStatement("int length0 = length0()")
-                .beginControlFlow("if (length0 != -1)")
-                .addStatement("valueRO.wrap(buffer, offset + FIELD_SIZE_LENGTH, length0)")
-                .endControlFlow()
-                .addStatement("return this")
-                .build();
-    }
-
-    private MethodSpec valueMethod()
-    {
-        return methodBuilder("value")
-                .addModifiers(PUBLIC)
-                .returns(DIRECT_BUFFER_TYPE)
-                .addStatement("return length0() == -1 ? null : valueRO")
-                .build();
-    }
-
-    private MethodSpec toStringMethod()
-    {
-        return methodBuilder("toString")
-                .addAnnotation(Override.class)
-                .addModifiers(PUBLIC)
-                .returns(String.class)
-                .addStatement("return String.format(\"\\\"%s\\\"\", asString())")
-                .build();
+            .addModifiers(PUBLIC, ABSTRACT)
+            .returns(String.class)
+            .build();
     }
 
     private MethodSpec length0Method()
     {
         return methodBuilder("length0")
-                .addModifiers(PRIVATE)
-                .returns(int.class)
-                .addStatement("int length = buffer().getByte(offset()) & 0xFF")
-                .addStatement("return length == 255 ? -1 : length")
-                .build();
+            .addModifiers(PUBLIC, ABSTRACT)
+            .returns(int.class)
+            .build();
     }
 
     private static final class BuilderClassBuilder
     {
         private final TypeSpec.Builder classBuilder;
         private final ClassName classType;
         private final ClassName stringType;
+        private final TypeName genericType;
 
         private BuilderClassBuilder(
             ClassName stringType,
             ClassName builderRawType)
         {
-            TypeName builderType = ParameterizedTypeName.get(builderRawType, stringType);
+            genericType = TypeVariableName.get("T", stringType);
```

Comment:
Suggest renaming `genericType` to `parameterType` because `T` is the type parameter to the generic builder type.